clients/src/main/java/org/apache/kafka/clients/admin/internals/AdminMetadataManager.java

```diff
@@ -226,20 +225,30 @@ public void transitionToUpdatePending(long now) {
         this.lastMetadataFetchAttemptMs = now;
     }
 
+    public void updateFailed(Throwable exception) {
+        // We depend on pending calls to request another metadata update
```

Comment:
Should we still update `lastMetadataUpdateMs` in `updateFailed()` so that we can follow `refreshBackoffMs` to avoid sending metadata request too fast? Also, should we update metadataFetchDelayMs() to backoff with`refreshBackoffMs` when state is UPDATE_REQUESTED?

Reply:
Ah, never mind. realized we should not `lastMetadataUpdateMs` here. `lastMetadataUpdateMs` is used with `metadataExpireMs` instead of `lastMetadataFetchAttemptMs`.

Reply:
Yes, that makes sense.

I was looking into how the admin client handles the backoff generally. Unless I'm missing something, it looks like the current retry logic inside `KafkaAdminClient#Call` does not respect the backoff between retries. I'll open a separate JIRA to fix this.